KomaMRICore/src/simulation/Flow.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    reset_magnetization!
+"""
+function reset_magnetization!(M::Mag{T}, Mxy::AbstractArray{Complex{T}}, motion::NoMotion{T}, t::AbstractArray{T}, ρ) where {T<:Real}
+   return nothing
+end
+
+function reset_magnetization!(M::Mag{T}, Mxy::AbstractArray{Complex{T}}, motion::MotionList{T}, t::AbstractArray{T}, ρ) where {T<:Real}
+   for m in motion.motions
+      t_unit = KomaMRIBase.unit_time(t, m.time)
+      idx = KomaMRIBase.get_idx(m.spins)
+      reset_magnetization!(@view(M[idx]), @view(Mxy[idx, :]), m.action, t_unit, @view(ρ[idx]))
+   end
+   return nothing
+end
+
+function reset_magnetization!(M::Mag{T}, Mxy::AbstractArray{Complex{T}}, action::KomaMRIBase.AbstractActionSpan{T}, t, ρ) where {T<:Real}
+   return nothing
+end
```

Comment:
I don't think you need to create two methods that `return nothing`. Just remove the type in `action/motion`. Is `t::AbstractArray{T}` needed?